src/screens/LegacyAccountBackup.js

```diff
@@ -0,0 +1,160 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+'use strict';
+
+import React, { useEffect } from 'react';
+import { AppState, ScrollView, StyleSheet, Text, View } from 'react-native';
+import { Subscribe } from 'unstated';
+
+import colors from '../colors';
+import fonts from '../fonts';
+import fontStyles from '../fontStyles';
+import AccountCard from '../components/AccountCard';
+import Background from '../components/Background';
+import Button from '../components/Button';
+import ScreenHeading from '../components/ScreenHeading';
+import TouchableItem from '../components/TouchableItem';
+import DerivationPasswordVerify from '../components/DerivationPasswordVerify';
+import AccountsStore from '../stores/AccountsStore';
+import { NetworkProtocols, NETWORK_LIST } from '../constants';
+import { alertBackupDone, alertCopyBackupPhrase } from '../util/alertUtils';
+
+export default class LegacyAccountBackup extends React.PureComponent {
+	render() {
+		return (
+			<Subscribe to={[AccountsStore]}>
+				{accounts => (
+					<LegacyAccountBackupView {...this.props} accounts={accounts} />
+				)}
+			</Subscribe>
+		);
+	}
+}
+
+function LegacyAccountBackupView(props) {
+	useEffect(() => {
+		const handleAppStateChange = nextAppState => {
+			if (nextAppState === 'inactive') {
+				props.navigation.goBack();
+			}
+		};
+
+		AppState.addEventListener('change', handleAppStateChange);
+		return function() {
+			const { accounts } = props;
+			const selectedKey = accounts.getSelectedKey();
+
+			if (selectedKey) {
+				accounts.lockAccount(selectedKey);
+			}
+
+			AppState.removeEventListener('change', handleAppStateChange);
+		};
+	}, []);
+
+	const { accounts, navigation } = props;
+	const { navigate } = navigation;
+	const isNew = navigation.getParam('isNew');
+	const {
+		address,
+		derivationPassword,
+		derivationPath,
+		name,
+		networkKey,
+		seed,
+		seedPhrase
+	} = isNew ? accounts.getNew() : accounts.getSelected();
+	const protocol =
```

Comment:
hmm this could be bad if protocol gets set to empty string. Could we throw an error and `<Alert />` for that case?

Reply:
Good catch! fallback to `NetworkProtocol.UNKNOW`